first-aid.lic

```diff
@@ -56,22 +57,56 @@ class FirstAid
   def study_charts(charts_to_read)
     charts_to_read
       .map { |_name, info| info }
-      .select { |info| info['scholarship'] <= DRSkill.getrank('Scholarship') }
+      .select { |info| info['scholarship'] <= effective_scholarship }
       .sort_by { |info| info['scholarship'] }
       .reverse
-      .take(20)
+      .take(@num_charts)
       .each do |info|
-        case bput("turn my #{@booktype} to #{info['index']}", 'You turn', 'That section does not exist', 'Turn what?')
+        case bput("turn my #{@booktype} to #{info['index']}", 'You turn', 'That section does not exist', 'Turn what?','almost impossible to do')
         when 'You turn'
-          case bput("study my #{@booktype}", 'you begin to study', 'you begin studying', 'With a sudden moment of', 'In a sudden moment of clarity', '^Why ', 'you continue to study', 'you attempt', 'you continue studying', 'You need to be holding')
+          case bput("study my #{@booktype}", 'gradually absorbing', 'difficult time comprehending the advanced text', 'suddenly makes sense to you', '^Why ', 'You need to be holding', 'discerned all you can')
+          when 'gradually absorbing'
+            loop{ message = bput("study my #{@booktype}", /Roundtime/, /makes sense/, /discerned all you can/ ); break if message == 'makes sense' || 'discerned all you can'}
```

Comment:
I think there might be a risk for infinite loop here. Would something like `5.times.do` instead of `loop` make more sense here?

Reply:
oddly it jumps out after just one retry. I've never had it even try a third but I don't see why we can't limit it just in case.

Reply:
Just being wary of something that can potentially loop forever given an edge case. Could be unfounded :)